File: spring-cloud-openfeign/feign-consumer/src/main/java/com/livk/consumer/biz/HttpUserRemoteService.java
package com.livk.consumer.biz;

import com.livk.autoconfigure.http.annotation.HttpProvider;
import com.livk.provider.api.domain.Users;
import org.springframework.web.service.annotation.GetExchange;

import java.util.List;

/**
 * @author livk
 */
@HttpProvider(value = "httpUserRemoteService", url = "http://feign-provider-biz")
public interface HttpUserRemoteService {

    @GetExchange("/users")
    List<Users> users();
}

File: spring-cloud-openfeign/feign-consumer/src/main/kotlin/com/livk/consumer/biz/HttpUserRemoteService.kt
package com.livk.consumer.biz

import com.livk.provider.api.domain.Users
import org.springframework.web.service.annotation.GetExchange
import org.springframework.web.service.annotation.HttpExchange

/**
 * @author livk
 */
@HttpExchange(value = "httpUserRemoteService", url = "http://feign-provider-biz")
interface HttpUserRemoteService {
    @GetExchange("/users")
    fun users(): List<Users>
}
